src/libraries/System.Private.CoreLib/src/System/Collections/Hashtable.cs

```diff
@@ -730,6 +757,9 @@ private void rehash(int newsize)
             //   2) Protect against an OutOfMemoryException while allocating this
             //      new bucket[].
             bucket[] newBuckets = new bucket[newsize];
+#if TARGET_64BIT
+            _fastModMultiplier = HashHelpers.GetFastModMultiplier((uint)newsize);
+#endif
```

Comment:
This is not safe.  One of the key reasons code still uses Hashtable is its threading guarantees, which enable any number of reads to execute concurrently with one writer.  By updating the _fastModMultiplifer here, concurrent readers could start to fail.